File: hamersapp/src/main/java/nl/ecci/hamers/beers/NewBeerActivity.java
package nl.ecci.hamers.beers;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.annotation.NonNull;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;

import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;
import java.util.HashMap;
import java.util.Map;

import nl.ecci.hamers.R;
import nl.ecci.hamers.helpers.DataManager;

public class NewBeerActivity extends AppCompatActivity {

    private SharedPreferences prefs;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.new_beer_activity);

        final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        final ActionBar actionBar = getSupportActionBar();

        prefs = PreferenceManager.getDefaultSharedPreferences(this);

        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeButtonEnabled(true);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                this.finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    public void postBeer(View view) {
        try {
            EditText beer_title = (EditText) findViewById(R.id.beer_title_et);
            EditText beer_picture = (EditText) findViewById(R.id.beer_picture_et);
            EditText beer_soort = (EditText) findViewById(R.id.beer_soort_et);
            EditText beer_percentage = (EditText) findViewById(R.id.beer_percentage_et);
            EditText beer_brewer = (EditText) findViewById(R.id.beer_brewer_et);
            EditText beer_country = (EditText) findViewById(R.id.beer_country_et);

            String title = URLEncoder.encode(beer_title.getText().toString(), "UTF-8");
            String picture = URLEncoder.encode(beer_picture.getText().toString(), "UTF-8");
            String soort = URLEncoder.encode(beer_soort.getText().toString(), "UTF-8");
            String percentage = URLEncoder.encode(beer_percentage.getText().toString(), "UTF-8");
            String brewer = URLEncoder.encode(beer_brewer.getText().toString(), "UTF-8");
            String country = URLEncoder.encode(beer_country.getText().toString(), "UTF-8");

            if (!percentage.contains("%")) {
                percentage = percentage + "%";
            }

            Map<String, String> params = new HashMap<>();
            params.put("beer[name]", title);
            params.put("beer[picture]", picture);
            params.put("beer[percentage]", percentage);
            params.put("beer[country]", country);
            params.put("beer[brewer]", brewer);
            params.put("beer[soort]", soort);

            DataManager.postData(this, prefs, DataManager.BEERURL, DataManager.BEERKEY, params);
        } catch (UnsupportedEncodingException ignored) {
        }
    }

    public void onSaveInstanceState(Bundle savedInstanceState) {
        super.onSaveInstanceState(savedInstanceState);
    }

    public void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
    }
}

File: hamersapp/src/main/java/nl/ecci/hamers/beers/NewBeerActivity.kt
package nl.ecci.hamers.beers

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.DatePicker
import com.android.volley.VolleyError
import kotlinx.android.synthetic.main.activity_new_item.*
import kotlinx.android.synthetic.main.stub_new_beer.*
import nl.ecci.hamers.R
import nl.ecci.hamers.helpers.DataUtils
import nl.ecci.hamers.helpers.NewItemActivity
import nl.ecci.hamers.loader.Loader
import nl.ecci.hamers.loader.PostCallback
import org.json.JSONException
import org.json.JSONObject

class NewBeerActivity : NewItemActivity() {

    private var beerID: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_new_item)

        initToolbar()

        stub_new_item.layoutResource = R.layout.stub_new_beer
        stub_new_item.inflate()

        beerID = intent.getIntExtra(Beer.BEER, -1)
        if (beerID != -1) {
            val beer = DataUtils.getBeer(this, beerID)
            beer_name.setText(beer.name)
            beer_picture.setText(beer.imageURL)
            beer_soort.setText(beer.kind)
            beer_percentage.setText(beer.percentage)
            beer_brewer.setText(beer.brewer)
            beer_country.setText(beer.country)

            title = "Wijzig " + beer.name
        }
    }

    override fun postItem() {
        var percentage = beer_percentage.text.toString()

        if (!percentage.contains("%")) {
            percentage += "%"
        }

        val body = JSONObject()
        try {
            body.put("name", beer_name!!.text.toString())
            body.put("picture", beer_picture.text.toString())
            body.put("percentage", percentage)
            body.put("country", beer_country.text.toString())
            body.put("brewer", beer_brewer.text.toString())
            body.put("soort", beer_soort.text.toString())
        } catch (ignored: JSONException) {
        }

        Loader.postOrPatchData(this, Loader.BEERURL, body, beerID, object : PostCallback {
            override fun onSuccess(response: JSONObject) {
                val returnIntent = Intent()
                returnIntent.putExtra(SingleBeerActivity.beerName, beer_name.text.toString())
                returnIntent.putExtra(SingleBeerActivity.beerKind, beer_soort.text.toString())
                returnIntent.putExtra(SingleBeerActivity.beerPercentage, beer_percentage.text.toString())
                returnIntent.putExtra(SingleBeerActivity.beerBrewer, beer_brewer.text.toString())
                returnIntent.putExtra(SingleBeerActivity.beerCountry, beer_country.text.toString())
                setResult(Activity.RESULT_OK, returnIntent)
                finish()
            }

            override fun onError(error: VolleyError) {
                disableLoadingAnimation()
            }
        })
    }

    override fun onDateSet(datePicker: DatePicker?, year: Int, month: Int, day: Int) {
    }
}